tests/unit/test_log_forwarder.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import textwrap
+import unittest
+from unittest.mock import MagicMock
+
+from charms.loki_k8s.v1.loki_push_api import LogForwarder
+from ops.charm import CharmBase
+from ops.testing import Harness
+
+
+class FakeCharm(CharmBase):
+    """Container charm for forwarding logs using the logforwarder class."""
+
+    metadata_yaml = textwrap.dedent(
+        """
+        containers:
+          consumer:
+            resource: consumer-image
+
+        requires:
+          logging:
+            interface: loki_push_api
+        """
+    )
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.log_forwarder = LogForwarder(self)
```

Comment:
we also need to test the 'non-default loki_push_api endpoint path'

perhaps parametrize? And, we need to test the get_loki_endpoint callback

What's the coverage of the lib?